### ISSUE TITLE
LInk is generated to the wrong endpoint

### ISSUE THREAD

**Issue by robinrosenberg** [stoicflame/enunciate#1056]
I tried to deprecate a method and added a link to the recommended method.
When the original method's name is a prefix to the recommended method, the link point back to the referring method, not the new one. If more than one method all(?) methods' links points to the method with the short name, even if the other methods' names are not prefixes of the firsts.

package bug;

```java
import javax.ws.rs.*;
import javax.ws.rs.core.Response;

@Path("bug")
public class Buggy {
    /**
     * The link points to this method, not the other method.
     *
     * <p>the name of the pointed to method has this method's name aa a prefix.
     *
     * @return response
     * @deprecated see {@link #m2()}
     */
    @Deprecated
    @POST
    @Consumes("application/xml")
    public Response m() {
        return null;
    }

    /**
     * The link point is not correct either but if #m is removed, it will be correct.
     * Removing this methods has no effect in the bad link in the first method.
     *
     * <p>The name of the pointed to method does not have this method's name aa a prefix.
     *
     * @return response
     * @deprecated see {@link #m2()}
     */
    @Deprecated
    @POST
    @Consumes("text/plain")
    public Response xx() {
        return null;
    }

    /** @return response*/
    @POST
    @Consumes("application/json")
    public Response m2() {
        return null;
    }
}
```

**Comment by stoicflame** [comment]
Okay, thanks for the report.

**Comment by stoicflame** [comment]
Fixed in `2.13.3`.
